src/blocks/helpers/prompt.ts

```diff
@@ -286,3 +286,51 @@ export function injectConversationIntoPrompt( embeddedPrompt: PromptData, conver
 		]
 	};
 }
+
+/**
+ * Injects content into a template. If no match is found, adds the content at the end.
+ *
+ * @param template The template to inject into.
+ * @param content The content to inject.
+ * @returns The template with the content injected or appended.
+ */
+export function tryInjectIntoTemplate( template: string, content: string ): string {
+	if ( ! template ) {
+		return content;
+	}
+
+	const injected = template.replace( /{text_input}/gi, () => content || '{text_input}' );
+
+	if ( injected === template && content ) {
+		return template + ' ' + content;
+	}
+
+	return injected;
```

Comment:
Maybe use a ternary operator here?

```js
return (injected === template && content) ? template + ' ' + content : injected;
``` 

I'm not 100% sure what the best practice is in JS land. What do you think @selul?

Reply:
hey,

I havent noticed or learned about a specific pattern regarding this.

I think if the expression is easier to understand this way, I would definitely advocate for it.

Reply:
@selul which expression just to be clear 😅

Reply:
Ah, the one using ternary operator.